### DISCUSSION TITLE
problem compiling gcc-4.8.5 on mbp 32bit

### DISCUSSION THREAD

**Discussion by Pippetta87** [igagis/esp-open-sdk#13] [Q&A] [answered]
Hi, trying to compile this package on macbookpro 32bits running debian but get this error:
`[ERROR]    /home/salvicio/src/esp-open-sdk/sdk/crosstool-NG/.build/src/gcc-4.8.5/gcc/reload1.c:89:24: error: use of an operand of type 'bool' in 'operator++' is forbidden in C++17
[ERROR]    gmake[4]: *** [Makefile:1058: reload1.o] Error 1
[ERROR]    gmake[3]: *** [Makefile:3889: all-gcc] Error 2
[ERROR]   
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Installing pass-2 core C gcc compiler'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@257]
[ERROR]  >>        called from: do_gcc_core_backend[scripts/build/cc/100-gcc.sh@545]
[ERROR]  >>        called from: do_gcc_core_pass_2[scripts/build/cc/100-gcc.sh@160]
[ERROR]  >>        called from: do_cc_core_pass_2[scripts/build/cc.sh@42]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@646]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      'share/doc/crosstool-ng/crosstool-ng-1.22.0-68-gfa9625cc/B - Known issues.txt'
[ERROR]   
[ERROR]  (elapsed: 43:59.71)
[44:01] / gmake[2]: *** [ct-ng:152 : build] Erreur 2
gmake[2] : on quitte le répertoire « /home/salvicio/src/esp-open-sdk/sdk/crosstool-NG »
make[1]: *** [../Makefile:152 : _toolchain] Erreur 2
make[1] : on quitte le répertoire « /home/salvicio/src/esp-open-sdk/sdk/crosstool-NG »
make: *** [Makefile:144 : crosstool-NG/.built] Erreur 2
`
Can someone help?
thanks
[config.log](https://github.com/esp-open-sdk/esp-open-sdk/files/8046400/config.log)
[build.log](https://github.com/esp-open-sdk/esp-open-sdk/files/8046401/build.log)

**Comment by ChrisMacGregor** [comment, marked as answer]
See https://github.com/esp-open-sdk/esp-open-sdk/issues/10 - the patch I mention there will fix this.  For the patch to work, you'll have to either start with a clean checkout and copy it to:

    sdk/crosstool-NG/patches/gcc/4.8.5/

or else copy it to both there and ALSO to:

    sdk/crosstool-NG/lib/crosstool-ng-1.22.0-60-g37b07f6f-dirty/patches/gcc/4.8.5/

(also, note that I'm working with the original pfalcon repo, not this one, so YMMV but it should be very similar).
